lib/py/src/protocol/fastbinary.c

```diff
@@ -1028,7 +1043,7 @@ decode_val(DecodeBuffer* input, TType type, PyObject* typeargs) {
     }
 
     len = readI32(input);
-    if (!check_ssize_t_32(len)) {
+    if (!check_list_length(len)) {
       return NULL;
```

Comment:
I guess we need to inform the user about reaching the limit somehow, e.g. via PyErr_SetString